### PR TITLE
Add Enemy support

### PR DESCRIPTION
**Pull request by 2767mr**
This PR adds support for rendering enemies as well as widgets for editing their properties.

Things that might need to be discussed:
* Boss (multi-part-) entities are broken but I cannot find out why. I decided to leave it in a semi-broken state because bosses are not common and do not need to be rendered perfectly in the first place.
* The  enemy widget has no clear way of showing where the enemy data begins and where it ends. This is mostly due to a lack of space. Any suggestions for improvements appreciated.
* Dropdowns do not seem to have the correct style. I left it as is because they are used in other places of the editor.

### CODE REVIEW

**Comment by Vegita2** [comment]
> The enemy widget has no clear way of showing where the enemy data begins and where it ends. This is mostly due to a lack of space. Any suggestions for improvements appreciated.

maybe something similar to how the crosscode devs did it? Basically showing another window similar to how npc states and events work. 

image

File: webapp/src/app/models/multi-entity-animation.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { Point3 } from './cross-code-map';
+import { TileSheet } from './tile-sheet';
+
+export interface MultiEntityAnimation {
+    DOCTYPE: 'MULTI_ENTITY_ANIMATION';
+
+    anims: Record<string, EntityAnim>;
+    baseSize: Point3;
+    namedSheets?: Record<string, TileSheet>;
```
Comment (Vegita2) [inline webapp/src/app/models/multi-entity-animation.ts:9]: isn't this type just `{ [key: string]: TileSheet }`?

File: webapp/src/styles.scss [hunk 2]
```diff
@@ -61,6 +61,26 @@ input[type=text].default-input, input:not([type]).default-input {
 	outline: solid 1px mat-color($mat-blue, 500);
 }
 
+
+.default-input.mat-select {
```
Comment (Vegita2) [inline webapp/src/styles.scss:65]: Where are these styles used?

**Review by Vegita2 — CHANGES_REQUESTED** [review]
Looks good.

Only thing is the added styles in `styles.scss`.
I'm not sure if they are still used or not

File: webapp/src/app/models/multi-entity-animation.ts (hunk 1 above)
Comment (2767mr) [reply webapp/src/app/models/multi-entity-animation.ts:9]: That is correct - `type Record<A, B> = { [key: A]: B }`. Should I change it?

File: webapp/src/styles.scss (hunk 2 above)
Comment (2767mr) [reply webapp/src/styles.scss:65]: They are used for selects inside widgets. Without these there are no borders and the size is slightly messed up. See https://github.com/CCDirectLink/crosscode-map-editor/blob/enemy/webapp/src/app/shared/widgets/string-widget/string-widget.component.html#L4-L10

File: webapp/src/app/models/multi-entity-animation.ts (hunk 1 above)
Comment (Vegita2) [reply webapp/src/app/models/multi-entity-animation.ts:9]: I don't really care what we use. I just never saw `Record`used before

File: webapp/src/styles.scss (hunk 2 above)
Comment (Vegita2) [reply webapp/src/styles.scss:65]: alright

**Review by Vegita2 — APPROVED** [review]

**Pull request merged**